### ISSUE TITLE
Latest SVN don't build

### ISSUE THREAD

**Issue by GoogleCodeExporter** [yuuues/xbt#35]
```
Hey,

Seems like after last change to build.cpp it don't build.

server.cpp:189: error: expected initializer before Б─≤:Б─≥ token
server.cpp:194: error: could not convert 
Б─≤m_torrents.boost::unordered_map<Key, T, Hash, Pred, Alloc>::erase [with 
Key = std::array<char, 20ul>, T = t_torrent, Hash = 
boost::hash<std::array<char, 20ul> >, Pred = std::equal_to<std::array<char, 
20ul> >, Alloc = std::allocator<std::pair<const std::array<char, 20ul>, 
t_torrent> 
>](boost::unordered_detail::hash_const_iterator<std::allocator<std::pair<const 
std::array<char, 20ul>, t_torrent> >, 
boost::unordered_detail::ungrouped>(((const 
boost::unordered_detail::hash_iterator<std::allocator<std::pair<const 
std::array<char, 20ul>, t_torrent> >, 
boost::unordered_detail::ungrouped>&)((const 
boost::unordered_detail::hash_iterator<std::allocator<std::pair<const 
std::array<char, 20ul>, t_torrent> >, boost::unordered_detail::ungrouped>*)(& 
i)))))Б─≥ to Б─≤boolБ─≥
server.cpp:195: error: expected primary-expression before Б─≤}Б─≥ 
token
server.cpp:195: error: expected Б─≤)Б─≥ before Б─≤}Б─≥ token
server.cpp:195: error: expected primary-expression before Б─≤}Б─≥ 
token
server.cpp:195: error: expected Б─≤;Б─≥ before Б─≤}Б─≥ token

```

Original issue reported on code.google.com by `erem...@gmail.com` on 8 Dec 2013 at 10:40

**Comment by GoogleCodeExporter** [comment]
```
G++ 4.6 is required
```

Original comment by `olafvdspek` on 9 Jan 2014 at 3:57
- Changed state: **Invalid**

**Comment by GoogleCodeExporter** [comment]
```
That is not possible on latest CentOS as provided version is 4.4, so support 
for that distro is out the window I guess
```

Original comment by `erem...@gmail.com` on 9 Jan 2014 at 11:42

**Comment by GoogleCodeExporter** [comment]
```
4.6 was released 3 years ago...
```

Original comment by `olafvdspek` on 12 Jan 2014 at 12:25

**Comment by GoogleCodeExporter** [comment]
```
Yeah, but you know enteriprse RHEL based distros don't upgrade stuff like GCC 
often so chose stability. Plenty of people use those distros on servers. Choice 
is yours obviously, RHEL 7 is in beta, so sposse we'll wait till that.
```

Original comment by `erem...@gmail.com` on 12 Jan 2014 at 12:31

**Comment by GoogleCodeExporter** [comment]
```
I know, but I don't like to wait forever. Debian released 7.0 with GCC 4.7 last 
year.

BTW, you could also install revision 2396.
```

Original comment by `olafvdspek` on 12 Jan 2014 at 12:37
